lib/galaxy/webapps/galaxy/api/jobs.py

```diff
@@ -257,148 +266,262 @@ def common_problems(self, trans: ProvidesUserContext, id, **kwd):
-
-    @expose_api_anonymous
-    def metrics(self, trans: ProvidesUserContext, **kwd):
+        return self.service.dictify_associations(trans, job.output_datasets, job.output_library_datasets)
+
+    @router.post(
+        "/api/jobs/{id}/error",
+        name="report_error",
+        summary="Submits a bug report via the API.",
+    )
+    def error(
+        self,
+        payload: Annotated[ReportJobErrorPayload, ReportErrorBody],
+        id: Annotated[DecodedDatabaseIdField, JobIdPathParam],
+        trans: ProvidesUserContext = DependsOnTrans,
+    ) -> JobErrorSummary:
+        # Get dataset on which this error was triggered
+        dataset_id = payload.dataset_id
+        dataset = self.service.hda_manager.get_accessible(id=dataset_id, user=trans.user)
+        # Get job
+        job = self.service.get_job(trans, id)
+        if dataset.creating_job.id != job.id:
+            raise exceptions.RequestParameterInvalidException("dataset_id was not created by job_id")
+        tool = trans.app.toolbox.get_tool(job.tool_id, tool_version=job.tool_version) or None
+        email = payload.email
+        if not email and not trans.anonymous:
+            email = trans.user.email
+        messages = trans.app.error_reports.default_error_plugin.submit_report(
+            dataset=dataset,
+            job=job,
+            tool=tool,
+            user_submission=True,
+            user=trans.user,
+            email=email,
+            message=payload.message,
+        )
+        return JobErrorSummary(messages=messages)
+
+    @router.get(
+        "/api/jobs/{id}/inputs",
+        name="get_inputs",
+        summary="Returns input datasets created by a job.",
+    )
+    def inputs(
+        self,
+        id: Annotated[DecodedDatabaseIdField, JobIdPathParam],
+        trans: ProvidesUserContext = DependsOnTrans,
+    ) -> List[JobAssociation]:
+        job = self.service.get_job(trans=trans, job_id=id)
+        return self.service.dictify_associations(trans, job.input_datasets, job.input_library_datasets)
+
+    @router.get(
+        "/api/jobs/{id}/outputs",
+        name="get_outputs",
+        summary="Returns output datasets created by a job.",
+    )
+    def outputs(
+        self,
+        id: Annotated[DecodedDatabaseIdField, JobIdPathParam],
+        trans: ProvidesUserContext = DependsOnTrans,
+    ) -> List[JobAssociation]:
+        job = self.service.get_job(trans=trans, job_id=id)
+        return self.service.dictify_associations(trans, job.output_datasets, job.output_library_datasets)
+
+    @router.get(
+        "/api/jobs/{id}/parameters_display",
+        name="resolve_parameters_display",
+        summary="Resolve parameters as a list for nested display.",
+    )
+    def parameters_display_by_job(
+        self,
+        id: Annotated[DecodedDatabaseIdField, JobIdPathParam],
+        hda_ldda: Annotated[DatasetSourceType, HdaLddaQueryParam] = DatasetSourceType.hda,
```

Comment:
This parameter doesn't make sense here, we already require a job id. can you make this optional and mark it as deprecated?

Reply:
Yes